java/src/jmri/jmrit/logix/Engineer.java

```diff
@@ -740,10 +767,13 @@ public void stopRun(boolean abort, boolean turnOffFunctions) {
         if (abort) {
             _abort =true;
         }
+        synchronized (this) {
+            notifyAll();
```

Comment:
Spotbugs failed on this line:
`Medium: Naked notify in jmri.jmrit.logix.Engineer.stopRun(boolean, boolean) [jmri.jmrit.logix.Engineer] At Engineer.java:[line 771] NN_NAKED_NOTIFY`

These two links might give some guidance about the problem:

https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html

> NN: Naked notify (NN_NAKED_NOTIFY)
> 
> A call to notify() or notifyAll() was made without any (apparent) accompanying modification to mutable object state.  In general, calling a notify method on a monitor is done because some condition another thread is waiting for has become true.  However, for the condition to be meaningful, it must involve a heap object that is visible to both threads.
> 
> This bug does not necessarily indicate an error, since the change to mutable object state may have taken place in a method which then called the method containing the notification.
> 

https://stackoverflow.com/questions/5193459/how-to-fix-naked-notify-warning-of-findbugs

> The naked notify warning means that using the notify() method implies that there is some other thread waiting for some mutable state to change, and is waiting to be notified. But your synchronized block did not modify any mutable state, and so it seems odd that you would need the notify. If you modified the state of an object outside of the synchronized block, then it seems dubious that this code is thread safe, as there is another thread that is reading this data